mirdata/web_downloader.py

```diff
@@ -0,0 +1,201 @@
+    Fetch a dataset pointed by remote's url, save into path using remote's
+    filename and ensure its integrity based on the MD5 Checksum of the
+    downloaded file.
+
+    Adapted from scikit-learn's sklearn.datasets.base._fetch_remote.
+
+    Parameters
+    -----------
+    remote: RemoteFileMetadata
+        Named tuple containing remote dataset meta information: url, filename
+        and checksum
+    data_home: string
+        Directory to save the file to.
+    force_overwrite: bool
+        If True, overwrite existing file with the downloaded file.
+        If False, does not overwrite, but checks that checksum is consistent.
+
+    Returns
+    -------
+    file_path: string
+        Full path of the created file.
+    """
+    download_path = os.path.join(data_home, remote.filename)
+    if not os.path.exists(download_path) or force_overwrite:
+        # If file doesn't exist or we want to overwrite, download it
+        with DownloadProgressBar(
+            unit='B', unit_scale=True, miniters=1, desc=remote.url.split('/')[-1]
+        ) as t:
+            try:
+                _download_large_file(remote.url, download_path, t.update_to)
+            except HTTPError:
+                error_msg = """
+                            mirdata failed to download the dataset!
+                            Please try again in a few minutes.
+                            If this error persists, please raise an issue at
+                            https://github.com/mir-dataset-loaders/mirdata,
+                            and tag it with 'broken-link'.
+                            """
+                raise HTTPError(error_msg)
+
+    checksum = md5(download_path)
+    if remote.checksum != checksum:
+        raise IOError(
+            '{} has an MD5 checksum ({}) '
+            'differing from expected ({}), '
+            'file may be corrupted.'.format(download_path, checksum, remote.checksum)
+        )
+    return download_path
+
+
+def download_zip_file(zip_remote, save_dir, force_overwrite, cleanup=False):
+    zip_download_path = download_from_remote(
+        zip_remote, save_dir, force_overwrite)
+    unzip(zip_download_path, save_dir, cleanup=cleanup)
+
+
+def unzip(zip_path, save_dir, cleanup=False):
+    """Unzip a zip file to a specified save location.
+
+    Parameters
+    ----------
+    zip_path: str
+        Path to zip file
+    save_dir: str
+        Path to save unzipped data
+    cleanup: bool, default=False
+        If True, remove zipfile after unzipping.
+    """
+    zfile = zipfile.ZipFile(zip_path, 'r')
+    zfile.extractall(save_dir)
+    zfile.close()
+    if cleanup:
+        os.remove(zip_path)
+
+
+def download_tar_file(tar_remote, save_dir, force_overwrite, cleanup=False):
+    tar_download_path = download_from_remote(
+        tar_remote, save_dir, force_overwrite)
+    untar(tar_download_path, save_dir, cleanup=cleanup)
+
+
+def untar(tar_path, save_dir, cleanup=False):
```

Comment:
This could also stay in utils since it's not web download specific.

Reply:
true. So far it's only used internally by other functions in web_download - do you still think it should move?